lib/rules/no-useless-lookaround-assertions.ts

```diff
@@ -0,0 +1,115 @@
+import type {
+    LookaheadAssertion,
+    LookaroundAssertion,
+    LookbehindAssertion,
+} from "regexpp/ast"
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+
+export default createRule("no-useless-lookaround-assertions", {
+    meta: {
+        docs: {
+            description: "disallow useless nested lookaround assertions",
+            category: "Best Practices",
+            // TODO Switch to recommended in the major version.
+            // recommended: true,
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [],
+        messages: {
+            unexpected: "This {{kind}} assertion is useless.",
+        },
+        type: "suggestion",
+    },
+    create(context) {
+        /**
+         * Create visitor
+         */
+        function createVisitor(
+            regexpContext: RegExpContext,
+        ): RegExpVisitor.Handlers {
+            return {
+                onAssertionEnter(aNode) {
+                    if (aNode.kind === "lookahead") {
+                        verifyForLookahead(regexpContext, aNode)
+                    } else if (aNode.kind === "lookbehind") {
+                        verifyForLookbehind(regexpContext, aNode)
+                    }
+                },
+            }
+        }
+
+        /** Verify for lookahead assertion */
+        function verifyForLookahead(
```

Comment:
Is it really necessary to have 2 methods for verifying? They only differ in which index they access and what `.kind` they require. A combined verify method could look like this:

```ts
        /** Verify for lookaround assertion */
        function verify(
            regexpContext: RegExpContext,
            assertion: LookaroundAssertion,
        ) {
            for (const alternative of assertion.alternatives) {
                const nested = alternative.elements.at(
                    assertion.kind === "lookahead" ? -1 : 0,
                )
                if (
                    nested?.type === "Assertion" &&
                    nested.kind === assertion.kind &&
                    !nested.negate
                ) {
                    reportLookaroundAssertion(regexpContext, nested)
                }
            }
        }
```